packages/opentelemetry-core/src/default_span.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Span,
+  SpanContext,
+  Status,
+  Attributes,
+  SpanKind,
+} from '@opentelemetry/types';
+
+/** This is a no-op implementation of Span. */
+export class DefaultSpan implements Span {
+  protected name = 'DefaultSpan'; // default
+  protected spanId = '';
+  protected parentSpanId = '';
+  protected kind = SpanKind.INTERNAL; // default
+  protected spanContext: SpanContext | null = null;
```

Comment:
How can the span context be null? I don't think this is possible in a real implementation, so it shouldn't be possible in the no-op either.

Reply:
What's your suggestion? I think other language added some invalid `SpanContext`.

Reply:
I would definitely add an invalid `SpanContext` so that callers can expect to always receive a span context when using `span.context()`. Otherwise this will result in a lot of unnecessary `null` checks.